torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -188,7 +195,8 @@ struct GraphFuser {
     // are not necessarily correct.
     if (node->owningBlock() != block_)
       return false;
-    return node->kind() == prim::FusionGroup || isSimpleMap(node);
+    return node->kind() == prim::FusionGroup ||
+        node->kind() == aten::_grad_sum_to_size || isSimpleMap(node);
```

Comment:
I'd really like us to avoid putting `_grad_sum_to_size` nodes in fusion groups unnecessarily. We should either have stronger checks for them (that adding them would in fact help us fuse more), or we should have a postprocessing pass that e.g. will move them out of the group if they are applied to inputs, or create outputs.

Reply:
Thanks for going through this!

So postprocessing or checking if the input of _grad_sum_to_size is fusible? Do you have a preference?
Would the "create outputs" comment be mitigated by the deduplication in the fuser itself?
Personally, I would envision that one would move cases where the sumtosizes of the outputs are "ascending" (i.e. you can sort the dimensions in a way that every tensor only has the summations last) into the kernel itself.

Reply:
So I now do this. This probably means we do not fuse some cases where we would like to - e.g. the milstm backward could be such a case, but we are doing better than before and maybe we just are conservative.